pkg/manager/manager.go

```diff
@@ -112,15 +146,28 @@ func RunManager(ctx context.Context, kubeConfig *rest.Config) error {
 		utils.ManagedByAddonManager,
 	)
 
+	addonTemplateController := addontemplate.NewAddonTemplateController(
+		kubeConfig,
+		hubKubeClient,
+		addonClient,
+		addonInformerFactory.Addon().V1alpha1().ClusterManagementAddOns(),
```

Comment:
Does this controller need a filter? Considering the case when cma is configured with addontemplate but claim managed by self by annotation `addon.open-cluster-management.io/lifecycle: "self"`

Reply:
addonConfigController and managementAddonConfigController should be considered to filter too in self case ?

Reply:
discussed with haoqing offline, the `utils.FilterOutTheBuiltInAddOnConfigGVRs(a.addonConfigs)` will do the filter. so I think self filter is not necessary.